Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/playbook-MDE_-_Host_Advanced_Hunting_For_Network_Activity.yml

```diff
@@ -1526,6 +1566,140 @@ tasks:
           "y": 1360
         }
       }
+  '45':
+    id: '45'
+    taskid: 88124d1e-6223-43c8-89b1-264076f46201
+    type: condition
+    task:
+      id: 88124d1e-6223-43c8-89b1-264076f46201
+      version: -1
+      name: Check if device information was provided
+      description: Check if device information was provided
+      type: condition
+      iscommand: false
+      brand: ''
+    nexttasks:
+      yes:
+      - '36'
+      - '39'
+    separatecontext: false
+    conditions:
+    - label: yes
+      condition:
+      - - operator: isNotEmpty
+          left:
+            value:
+              simple: inputs.DeviceName
+            iscontext: true
+        - operator: isNotEmpty
+          left:
+            value:
+              simple: inputs.DeviceID
+            iscontext: true
+    view: |-
+      {
+        "position": {
+          "x": 2180,
+          "y": 510
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  '46':
+    id: '46'
+    taskid: 3b430e7e-7a24-4e44-8a7c-2938d89f1850
+    type: condition
+    task:
+      id: 3b430e7e-7a24-4e44-8a7c-2938d89f1850
+      version: -1
+      name: Are DeviceID and FileName provided?
+      description: Are DeviceID and FileName provided?
+      type: condition
+      iscommand: false
+      brand: ''
+    nexttasks:
+      '#default#':
+      - '29'
+      yes:
+      - '9'
+    separatecontext: false
+    conditions:
+    - label: yes
+      condition:
+      - - operator: isNotEmpty
+          left:
+            value:
+              simple: inputs.FileSha256
+            iscontext: true
+      - - operator: isNotEmpty
+          left:
+            value:
+              simple: inputs.DeviceID
+            iscontext: true
+    view: |-
+      {
+        "position": {
+          "x": 990,
+          "y": 860
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  '48':
+    id: '48'
+    taskid: fd8e6713-5106-4527-82c0-dc6970184e36
+    type: condition
+    task:
+      id: fd8e6713-5106-4527-82c0-dc6970184e36
+      version: -1
+      name: Are DeviceID and FileName provided?
```

Comment:
I think you can add those condition one time (at the start of the playbook) with 1 new task and not 3 that checking the same inputs.